lab_2/interpolation/intrepolation_module.py

```diff
@@ -0,0 +1,108 @@
+from interpolation.increasing_counts import IncreasingCounts
+from interpolation.read_dataset import ReadData
+from interpolation.interpolation_method import Interpn, RegularGridInterpolator, RectBivariateSpline
+import cv2
+import numpy as np
+import copy
+import itertools
+import json
+
+
+
+class InterpolationModule:
+    def __init__(self, image: np.array, scale: int):
+        self._image = image
+        self._scale = scale
+
+    def interpolation(self):
+        images = copy.copy(self._image.images)
+        increase_data = IncreasingCounts(images, self._scale)
+        for i in range(len(increase_data.increase_image)):
+            cv2.imwrite(f'../increase_dataset/{self._scale}/increase_dataset{i}.png', increase_data.increase_image[i])
+            interpn_data = Interpn.interpolation(increase_data.increase_image[i], data.images[i], self._scale)
+            cv2.imwrite(f'../interpolation_dataset/interpn/{self._scale}/im{i}.png', interpn_data)
+            rgi_data = RegularGridInterpolator.interpolation(increase_data.increase_image[i], data.images[i],
+                                                             self._scale)
+            cv2.imwrite(f'../interpolation_dataset/RGI/{self._scale}/im{i}.png', rgi_data.values)
+            rbs_data = RectBivariateSpline.interpolation(increase_data.increase_image[i], data.images[i], self._scale)
+            cv2.imwrite(f'../interpolation_dataset/RBS/{self._scale}/im{i}.png', rbs_data)
+
+
+if __name__ == '__main__':
+    path = "../dataset/x0.3333"
+    data = ReadData(path)
+    increase = 3
+    interpolation_data = InterpolationModule(data, increase)
+    interpolation_data.interpolation()
+    orig_img = ReadData('shift_orig_img')
+    for j in range(2, 11):
+        interp_data = ReadData(f'../interpolation_dataset/RBS/{j}')
+        increase_data = ReadData(f'../increase_dataset/{j}')
```

Comment:
Можно вынести чтение данных в отдельную функцию `load_data`, это улучшит читаемость, а также уменьшит дублирование кода.
```suggestion
       def load_data(j):
                 interp_data = ReadData(f'../interpolation_dataset/RBS/{j}')
                 increase_data = ReadData(f'../increase_dataset/{j}')
       return interp_data, increase_data
```